android/beagle/src/main/java/br/com/zup/beagle/android/components/refresh/PullToRefresh.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.components.refresh
+
+import android.graphics.Color
+import android.view.View
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import br.com.zup.beagle.android.action.Action
+import br.com.zup.beagle.android.context.Bind
+import br.com.zup.beagle.android.context.ContextComponent
+import br.com.zup.beagle.android.context.ContextData
+import br.com.zup.beagle.android.context.valueOfNullable
+import br.com.zup.beagle.android.utils.handleEvent
+import br.com.zup.beagle.android.utils.observeBindChanges
+import br.com.zup.beagle.android.view.ViewFactory
+import br.com.zup.beagle.android.widget.RootView
+import br.com.zup.beagle.android.widget.WidgetView
+import br.com.zup.beagle.annotation.RegisterWidget
+import br.com.zup.beagle.core.ServerDrivenComponent
+import br.com.zup.beagle.core.SingleChildComponent
+
+@RegisterWidget("pullToRefresh")
+data class PullToRefresh constructor(
+    override val context: ContextData? = null,
+    val onPull: List<Action>,
+    val isRefreshing: Bind<Boolean>? = null,
+    val color: Bind<String>? = null,
+    override val child: ServerDrivenComponent,
+) : WidgetView(), ContextComponent, SingleChildComponent {
+
+    constructor(
+        context: ContextData?,
```

Comment:
This constructor must have a value default for the context, otherwise the user will be forced to pass one, even if it is null.
```suggestion
        context: ContextData? = null,
```